sslr-core/src/main/java/com/sonar/sslr/api/typed/NodeBuilder.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SonarSource Language Recognizer
+ * Copyright (C) 2010 SonarSource
+ * dev@sonar.codehaus.org
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package com.sonar.sslr.api.typed;
+
+import com.google.common.annotations.Beta;
+import com.sonar.sslr.api.Rule;
+import com.sonar.sslr.api.TokenType;
+import com.sonar.sslr.api.Trivia;
+import org.sonar.sslr.grammar.GrammarRuleKey;
+
+import java.util.List;
+
+@Beta
+public interface NodeBuilder {
```

Comment:
Can we somehow check the consistency with `GrammarBuilder<N, T>` by making this interface also generic?